### ISSUE TITLE
NavigationView Crash in SDK 20 and 21

### ISSUE THREAD

**Issue by billyking991** [mapbox/mapbox-navigation-android#1409]
The code for creating routes, route options, and starting the navigation has not changed in my project, yet in Navigation 0.20.0 and 0.21.0, the app now crashes as soon as the route is calculated and the navigation starts in the NavigationView.

Console Logs:
```
FATAL EXCEPTION: main java.lang.NoSuchFieldError: no "Lcom/mapbox/geojson/Point;" field "location" in class "Lcom/mapbox/navigator/FixLocation;" or its superclasses
                      at com.mapbox.navigator.Navigator.updateLocation(Native Method)
                      at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigator.a(MapboxNavigator.java:36)
                      at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorBackgroundThread.a(RouteProcessorBackgroundThread.java:60)
                      at com.mapbox.services.android.navigation.v5.navigation.NavigationLocationEngineListener.onLocationChanged(NavigationLocationEngineListener.java:31)
                      at com.mapbox.services.android.navigation.v5.navigation.NavigationLocationEngineUpdater.a(NavigationLocationEngineUpdater.java:34)
                      at com.mapbox.services.android.navigation.v5.navigation.NavigationService.a(NavigationService.java:68)
                      at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation.onServiceConnected(MapboxNavigation.java:726)
                      at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1386)
                      at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1403)
                      at android.os.Handler.handleCallback(Handler.java:739)
                      at android.os.Handler.dispatchMessage(Handler.java:95)
                      at android.os.Looper.loop(Looper.java:135)
                      at android.app.ActivityThread.main(ActivityThread.java:5422)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at java.lang.reflect.Method.invoke(Method.java:372)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:914)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:707)
```

**Android API:**
22 - 27
**Mapbox Navigation SDK version:**
This issue did not happen with the following SDK versions:
```
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation:0.16.0'
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation:0.18.0'
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation:0.19.0'
```
It does happen in the following SDK versions:
```
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation:0.20.0'
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation:0.21.0'
```

### Steps to trigger behavior
1.  Calculate the route between origin and destination.
2.  Create the navigationViewOptions.
3.  startNavigation in the navigationView
4.  Crash

### Expected behavior
No crashes. LOL.
### Actual behavior
As soon as the the navigation starts, there is a crash (See previous debug output).  The app has been working fine up until SDK 20.  Any ideas why this may be happening?  Is there something I needed to add for SDK 20 that it might be missing?

Code:
```
NavigationRoute.builder(mContext)
                    .accessToken(Mapbox.getAccessToken())
                    .origin(origin)
                    .destination(destination)
                    .profile(navTypeFormat)
                    .exclude(EXCLUDE_TOLL)
                    .language(Locale.ENGLISH)
                    .build()
                    .getRoute(new Callback<DirectionsResponse>() {
                        @Override
                        public void onResponse(Call<DirectionsResponse> call, Response<DirectionsResponse> response) {
               
                            currentRoute = response.body().routes().get(0);

                            NavigationViewOptions.Builder navigationViewOptions = NavigationViewOptions.builder();
                            navigationViewOptions.directionsRoute(currentRoute);
                            navigationViewOptions.navigationListener(NavigationActivity.this);
                            navigationViewOptions.milestoneEventListener(NavigationActivity.this);
                            navigationViewOptions.speechAnnouncementListener(NavigationActivity.this);
                            navigationViewOptions.bannerInstructionsListener(NavigationActivity.this);
                            navigationViewOptions.routeListener(NavigationActivity.this);
                            navigationViewOptions.shouldSimulateRoute(false);

                            navigationView.startNavigation(navigationViewOptions.build());
                        }

                        @Override
                        public void onFailure(Call<DirectionsResponse> call, Throwable throwable) {
                            showNoRoutesAlert();
                        }
                    });
```

**Comment by danesfeder** [comment]
Hey @billyking991, thanks for reporting this.  This code you are referencing was modified starting with `0.20.0`, we will dig into the root cause and post back here after some investigation.

**Comment by billyking991** [comment]
Thank you so much!  I appreciate the quick response.  I can always use 19 in the meantime.  I'm just trying to keep the SDK as current as possible as you make changes and improvements.

**Comment by danesfeder** [comment]
@billyking991 awesome thank you, noting here that this may be ProGuard related and we may need to adjust our rules to resolve this.

**Comment by billyking991** [comment]
Ahhh...we if you know what I need to add, please let me know and I'll add it and test.  You may be right about this.  I always forget about Proguard rules.  I haven't changed the pro guard rules for Mapbox since the creation of the project.

**Comment by billyking991** [comment]
I just wanted to check up on this.  No worries if you are still working on it.  I've been trying to change my ProGuard rules looking for a possible candidate, but haven't found anything.  In a pinch, I can revert back to 0.19.0, but I would eventually like to bring it up to the latest.  If you need more information, please let me know and I'll post it.  Thank you!

**Comment by danesfeder** [comment]
@billyking991 yeah we are still looking into this, sorry about that - do you mind posting your rules related to Mapbox so we can take a look and try to reproduce on our end?  Thanks!

**Comment by billyking991** [comment]
No worries at all.  I believe this is what I've added for Mapbox.  My ProGuard files are a bit of a mess.

I've seen the ProGuard files on GitHub.  Do you have a link to the official ProGuard file so I can compare on my end?

-dontnote retrofit2.Platform
-dontnote retrofit2.Platform$IOS$MainThreadExecutor
-dontwarn retrofit2.Platform$Java8
-keepattributes Signature
-keepattributes Exceptions
-keep class com.segment.analytics.** { *; }
-keep class com.mapbox.mapboxandroiddemo.model.usermodel.** { *; }
-keep class com.mapbox.services.api.directionsmatrix.v1.models.** { *; }
-dontwarn com.mapbox.api.staticmap.v1.models.StaticMarkerAnnotation
-dontwarn com.mapbox.api.staticmap.v1.models.StaticPolylineAnnotation

**Comment by danesfeder** [comment]
@billyking991 do you mind removing the Mapbox-related rules and see if this does anything?  We may need to make some adjustments to the rules we include on our side.  Thanks for your patience!

**Comment by billyking991** [comment]
No worries at all!  I've removed the rules and am getting the same crash.  Here is the log from the console:

FATAL EXCEPTION: main
                  java.lang.NoSuchFieldError: no "Lcom/mapbox/geojson/Point;" field "location" in class "Lcom/mapbox/navigator/FixLocation;" or its superclasses
                      at com.mapbox.navigator.Navigator.updateLocation(Native Method)
                      at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigator.a(MapboxNavigator.java:36)
                      at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorBackgroundThread.a(RouteProcessorBackgroundThread.java:60)
                      at com.mapbox.services.android.navigation.v5.navigation.NavigationLocationEngineListener.onLocationChanged(NavigationLocationEngineListener.java:31)
                      at com.mapbox.services.android.navigation.v5.navigation.NavigationLocationEngineUpdater.a(NavigationLocationEngineUpdater.java:34)
                      at com.mapbox.services.android.navigation.v5.navigation.NavigationService.a(NavigationService.java:68)
                      at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation.onServiceConnected(MapboxNavigation.java:726)
                      at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1658)
                      at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1687)
                      at android.os.Handler.handleCallback(Handler.java:789)
                      at android.os.Handler.dispatchMessage(Handler.java:98)
                      at android.os.Looper.loop(Looper.java:164)
                      at android.app.ActivityThread.main(ActivityThread.java:6940)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)

**Comment by danesfeder** [comment]
Hey @billyking991 👋 we may have found the source of the progaurd issues in https://github.com/mapbox/mapbox-navigation-android/issues/1452.  Do you mind testing the same?  Thanks!

**Comment by billyking991** [comment]
Yes!!!!  That fixes it!!  Please thank the person that found it!!  I can update to the latest and begin testing.  Thank you for checking into this for me.  I really appreciate it.  I'll let you know if I have any other issues like this.  By the way, nice work on the updates.  I've noticed some really nice stability changes.

**Comment by danesfeder** [comment]
@billyking991 awesome thank you!  We opened https://github.com/mapbox/mapbox-navigation-android/pull/1454 to add this to our rules so you won't have to include them in your own.  Should be in the next release this week.

**Comment by billyking991** [comment]
Perfect!  I look forward to v.0.22.  Now, I have to see what extra features I can use.  Thanks for the update.

**Comment by Guardiola31337** [comment]
Hey @billyking991 👋 friendly reminder that #1454 landed and it's available in the recent [`0.22.0`](https://github.com/mapbox/mapbox-navigation-android/releases/tag/v0.22.0) release version of the SDK 🎉 

Thanks for bringing this to us!
